src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -18,8 +18,8 @@
  * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
  */
 
-#include "gpio_driver.h"
-#include "gpio_nif.h"
+#include <sdkconfig.h>
+#if defined(CONFIG_AVM_ENABLE_GPIO_PORT_DRIVER) || defined(CONFIG_AVM_ENABLE_GPIO_NIFS)
```

Comment:
My preference is that this macro guard should only be around the REGISTER_ functions, so that the bulk of the code still gets compiled, but might not get linked in, if the KConfig is turned off.  That seems more similar to the IDF SDK behavior, which builds :allthethings: but only links what it needs.